wrangler-proto/src/main/java/io/cdap/wrangler/proto/ErrorRecordsException.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler.proto;
+
+import io.cdap.wrangler.api.ErrorRecordBase;
+import java.util.List;
+
+/**
+ * An exception that contains error records.
+ */
+public class ErrorRecordsException extends RuntimeException {
+
+  private final List<ErrorRecordBase> errorRecords;
+
+  public ErrorRecordsException(List<ErrorRecordBase> errorRecords) {
+    super(String.format("%d error records were produced", errorRecords.size()));
```

Comment:
With the assumption that nothing in the UI is changing, it makes sense to also include the message of the first error in the exception message (or the first X errors, or the first N characters?), otherwise there will be no way to know what to correct. I would just avoid showing them all, as it could easily become a huge message that covers the entire screen.

I think if we were to change the UI, we would also want to do the full solution in the backend where all errors are collected and returned, rather than this intermediate solution.